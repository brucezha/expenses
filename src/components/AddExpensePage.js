import React from 'react';
import Button from '@material-ui/core/Button';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { Link } from 'react-router-dom';
import { startAddExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startAddExpense(expense);
        this.props.history.push('/');
    };
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Add Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        onSubmit={this.onSubmit}
                        cards={this.props.cards}
                    />
                    <Button component={Link} to="/" variant="contained" style={{ fontSize: '15px', textTransform: 'none' }} color="default">Cancel </Button>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => ({
    cards: state.cards
  });

const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(mapStateToProps, mapDispatchToProps)(AddExpensePage);