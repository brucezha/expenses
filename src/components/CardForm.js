import React from 'react';
import { Button } from '@material-ui/core';

export default class ExpenseForm extends React.Component {
    constructor(props) {
        // console.log(props.card.cardName);
        super(props);
        this.state = {
            cardName: props.card ? props.card.cardName : ''
        };
    }
    onCardNameChange = (e) => {
        const cardName = e.target.value;
        this.setState(() => ({ cardName }))
    };
    onSubmit = (e) => {
        e.preventDefault();

        if (!this.state.cardName) {
            //set error message
            this.setState(() => ({ error: 'Please enter a card name' }))
        } else {
            this.setState(() => ({ error: '' }));
            this.props.onSubmit({
                cardName: this.state.cardName,
            });
        }
    };
    render() {
        return (
            <form className="form" onSubmit={this.onSubmit}>
                {this.state.error && <p className="form__error">{this.state.error}</p>}
                <input
                    type="text"
                    placeholder="Enter the card you want to add"
                    autoFocus
                    className="text-input"
                    value={this.state.cardName}
                    onChange={this.onCardNameChange}
                />
                <div>
                    <Button type="submit" variant="contained" style={{ fontSize: '15px' }} color="primary">Submit Card</Button>
                </div>
            </form>
        )
    }
}