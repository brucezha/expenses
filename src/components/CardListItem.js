import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const CardListItem = ({ id, cardName }) => (
    <Link className="list-item" to={`/cards/${id}`}>
        <div>
            <h3 className="list-item__title">{cardName}</h3>
        </div>
        {/*
        <h3 className="list-item__data"> {card} </h3>
        <h3 className="list-item__data"> {numeral(amount / 100).format('$0,0.00')} </h3>
        */}
    </Link>
);

export default CardListItem;