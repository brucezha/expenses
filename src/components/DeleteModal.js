import React from 'react';
import Modal from 'react-modal';
import { Button } from '@material-ui/core';

const DeleteModale = (props) => (
    <Modal
        isOpen={!!props.deleteSelected}
        onRequestClose={props.handleClearDelete}
        ariaHideApp={false}
        closeTimeoutMS={400}
        className="modal"
    >
        <h3 className="modal__title">Are you sure you want to delete this entry?</h3>
        <div className="modal__body">
            <Button onClick={props.remove} variant="contained" style={{ fontSize: '15px' }} color="secondary">Yes</Button>
        &nbsp;
        &nbsp;
        <Button onClick={props.handleClearDelete} variant="contained" style={{ fontSize: '15px' }} color="primary">Actually No</Button>
        </div>
    </Modal>
);
export default DeleteModale;