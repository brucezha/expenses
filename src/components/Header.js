import { Button } from '@material-ui/core';
import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
    <header className="header-background">
        <div className="content-container">
            <div className="header__content">
                <Link className="header__title" to="/dashboard">
                    <h1>Budget</h1>
                </Link>
                <Button onClick={startLogout} style={{ fontSize: '15px', color: 'white', textTransform: 'none' }}>Logout</Button>
            </div>
        </div>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);