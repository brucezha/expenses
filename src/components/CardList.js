import React from 'react';
import Button from '@material-ui/core/Button';
import { connect } from 'react-redux';
import CardListItem from './CardListItem';
import { Link } from 'react-router-dom'
// import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Manage Cards</h1>
                <div className="page-header__actions">
                    <Button component={Link} to="/addCard" variant="contained" style={{ fontSize: '15px', textTransform: 'none' }} color="primary">Add A New Card</Button>
                </div>
            </div>
        </div>
        <div className="content-container">
            <div className="list-header">
                <div className="show-for-desktop">Card</div>
                <div className="show-for-desktop">Posted Charges</div>
            </div>
            <div className="list-body">
                {
                    props.cards.length === 0 ? (
                        <div className="list-item list-item--message">
                            <span>No card to show</span>
                        </div>
                    ) : (
                            props.cards.map((card) => {
                                return <CardListItem key={card.id} {...card} />
                            })
                        )
                }
            </div>
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: state.expenses,
        cards: state.cards
    };
};

export default connect(mapStateToProps)(ExpenseList);