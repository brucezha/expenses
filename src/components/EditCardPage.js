import { Button } from '@material-ui/core';
import React from 'react';
import { connect } from 'react-redux';
// import { startEditCard, startRemoveExpense } from '../actions/cards';
import { startEditCard } from '../actions/cards';
import CardForm from './CardForm';
import DeleteModal from './DeleteModal';

export class EditCardPage extends React.Component {
  state = {
    deleteSelected: false
  };
  handleCancel = () => {
    this.props.history.push('/');
  }
  handleDelete = () => {
    this.setState(() => ({ deleteSelected: true }));
  };
  handleClearDelete = () => {
    this.setState(() => ({ deleteSelected: false }));
  };
  onSubmit = (card) => {
    this.props.startEditCard(this.props.card.id, card);
    this.props.history.push('/manage');
  };
//   onRemove = () => {
//     this.props.startRemoveExpense({ id: this.props.expense.id });
//     this.props.history.push('/');
//   }
  render() {
    // console.log(this.props);
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Card</h1>
          </div>
        </div>
        <div className="content-container">
          <CardForm
            card ={this.props.card}
            onSubmit={this.onSubmit}
          />
          {/* 
          <Button
            onClick={this.handleCancel}
            variant="contained"
            style={{ fontSize: '15px' }}
            color="default">
            Cancel
        </Button>
        &nbsp;&nbsp;&nbsp;
          <Button
            onClick={this.handleDelete}
            variant="contained"
            style={{ fontSize: '15px' }}
            color="secondary">
            Remove
          </Button>
          <DeleteModal
            deleteSelected={this.state.deleteSelected}
            handleClearDelete={this.handleClearDelete}
            remove={this.onRemove}
          />
          */}
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  card: state.cards.find((card) => card.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
    startEditCard: (id, card) => dispatch(startEditCard(id, card)),
//   startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditCardPage);