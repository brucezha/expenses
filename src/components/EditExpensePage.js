import { Button } from '@material-ui/core';
import React from 'react';
import { connect } from 'react-redux';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';
import DeleteModal from './DeleteModal';

export class EditExpensePage extends React.Component {
  state = {
    deleteSelected: false
  };
  handleCancel = () => {
    this.props.history.push('/');
  }
  handleDelete = () => {
    this.setState(() => ({ deleteSelected: true }));
  };
  handleClearDelete = () => {
    this.setState(() => ({ deleteSelected: false }));
  };

  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  }
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <Button
            onClick={this.handleCancel}
            variant="contained"
            style={{ fontSize: '15px' }}
            color="default">
            Cancel
        </Button>
        &nbsp;&nbsp;&nbsp;
          <Button
            onClick={this.handleDelete}
            variant="contained"
            style={{ fontSize: '15px' }}
            color="secondary">
            Remove
          </Button>
          <DeleteModal
            deleteSelected={this.state.deleteSelected}
            handleClearDelete={this.handleClearDelete}
            remove={this.onRemove}
          />
          {/* <Button onClick={this.onRemove} variant="contained" style={{ fontSize: '15px' }} color="secondary">Remove</Button> */}
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);