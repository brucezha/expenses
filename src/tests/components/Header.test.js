import { Header } from '../../components/Header';
import React from 'react';
import { shallow } from 'enzyme';
import { Button } from '@material-ui/core';

test('should render Header correctly', () => {
    const wrapper = shallow(<Header startLogout={() => { }} />);
    expect(wrapper).toMatchSnapshot();
});

test('should call startLogout on button click', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<Header startLogout={startLogout} />);
    wrapper.find(Button).simulate('click');
    expect(startLogout).toHaveBeenCalled();
});