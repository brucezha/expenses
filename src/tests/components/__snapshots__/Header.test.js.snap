// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header-background"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        to="/dashboard"
      >
        <h1>
          Budget
        </h1>
      </Link>
      <WithStyles(ForwardRef(Button))
        onClick={[Function]}
        style={
          Object {
            "color": "white",
            "fontSize": "15px",
            "textTransform": "none",
          }
        }
      >
        Logout
      </WithStyles(ForwardRef(Button))>
    </div>
  </div>
</header>
`;
